lib/core/request.js

```diff
@@ -79,7 +80,15 @@ class Request {
     } else if (typeof body === 'string') {
       this.body = body.length ? Buffer.from(body) : null
     } else {
-      throw new InvalidArgumentError('body must be a string, a Buffer or a Readable stream')
+      try {
+        this.body = Readable.from(typeof body === 'function' ? body() : body)
```

Comment:
I think we might want to skip generator support and only to iterable. Generators have some semantics we might want to give more consideration to.

Reply:
Okay I'll remove generator support